src/mlpack/core/data/split_data.hpp

```diff
@@ -52,72 +149,35 @@ namespace data {
  * @param shuffleData If true, the sample order is shuffled; otherwise, each
  *     sample is visited in linear order. (Default true.)
  */
-template<typename T, typename U>
+template<typename T, typename LabelsType,
+         typename = std::enable_if_t<arma::is_arma_type<LabelsType>::value> >
 void StratifiedSplit(const arma::Mat<T>& input,
-                     const arma::Row<U>& inputLabel,
+                     const LabelsType& inputLabel,
                      arma::Mat<T>& trainData,
                      arma::Mat<T>& testData,
-                     arma::Row<U>& trainLabel,
-                     arma::Row<U>& testLabel,
+                     LabelsType& trainLabel,
+                     LabelsType& testLabel,
                      const double testRatio,
                      const bool shuffleData = true)
 {
-  /**
-   * Basic idea:
-   * Let us say we have to stratify a dataset based on labels:
-   * 0 0 0 0 0 (5 0s)
-   * 1 1 1 1 1 1 1 1 1 1 1 (11 1s)
-   *
-   * Let our test ratio be 0.2.
-   * Then, the number of 0 labels in our test set = floor(5 * 0.2) = 1.
-   * The number of 1 labels in our test set = floor(11 * 0.2) = 2.
-   *
-   * In our first pass over the dataset,
-   * We visit each label and keep count of each label in our 'labelCounts' uvec.
-   *
-   * We then take a second pass over the dataset.
-   * We now maintain an additional uvec 'testLabelCounts' to hold the label
-   * counts of our test set.
-   *
-   * In this pass, when we encounter a label we check the 'testLabelCounts' uvec
-   * for the count of this label in the test set.
-   * If this count is less than the required number of labels in the test set,
-   * we add the data to the test set and increment the label count in the uvec.
-   * If this count is equal to or more than the required count in the test set,
-   * we add this data to the train set.
-   *
-   * Based on the above steps, we get the following labels in the split set:
-   * Train set (4 0s, 9 1s)
-   * 0 0 0 0
-   * 1 1 1 1 1 1 1 1 1
-   *
-   * Test set (1 0s, 2 1s)
-   * 0
-   * 1 1
-   */
```

Comment:
Any particular reason to remove this comment?  I found it useful when I originally reviewed the function.

Reply:
I think this one was removed by @Abilityguy in one of his commits. @Abilityguy can you please explain why you decided to remove it ?

Reply:
I felt that the comment was too detailed to explain the function. But if it is helpful, then we can add it back again.

Reply:
In my opinion, it is always good to have documentation about the implementation. It really helps in understanding the codebase. I will add it again 👍